impl/monitor/monitor.go

```diff
@@ -0,0 +1,247 @@
+	t := time.NewTicker(s.pollPeriod)
+	for now := range t.C {
+		glog.Infof("Polling: %v", now)
+		ctx, _ := context.WithTimeout(context.Background(), 5*time.Second)
+		if _, err := s.pollMutations(ctx); err != nil {
+			glog.Errorf("pollMutations(_): %v", err)
+		}
+	}
+	return nil
+}
+
+// GetSignedMapRoot returns the latest valid signed map root the monitor
+// observed. Additionally, the response contains additional data necessary to
+// reproduce errors on failure.
+//
+// Returns the signed map root for the latest epoch the monitor observed. If
+// the monitor could not reconstruct the map root given the set of mutations
+// from the previous to the current epoch it won't sign the map root and
+// additional data will be provided to reproduce the failure.
+func (s *Server) GetSignedMapRoot(ctx context.Context, in *ktpb.GetMonitoringRequest) (*ktpb.GetMonitoringResponse, error) {
+	if len(s.proccessedSMRs) > 0 {
+		return s.proccessedSMRs[len(s.proccessedSMRs)-1], nil
+	}
+	return nil, ErrNothingProcessed
+}
+
+// GetSignedMapRootStream is a streaming API similar to GetSignedMapRoot.
+func (s *Server) GetSignedMapRootStream(in *ktpb.GetMonitoringRequest, stream mspb.MonitorService_GetSignedMapRootStreamServer) error {
+	// TODO(ismail): implement stream API
+	return grpc.Errorf(codes.Unimplemented, "GetSignedMapRootStream is unimplemented")
+}
+
+// GetSignedMapRootByRevision works similar to GetSignedMapRoot but returns
+// the monitor's result for a specific map revision.
+//
+// Returns the signed map root for the specified epoch the monitor observed.
+// If the monitor could not reconstruct the map root given the set of
+// mutations from the previous to the current epoch it won't sign the map root
+// and additional data will be provided to reproduce the failure.
+func (s *Server) GetSignedMapRootByRevision(ctx context.Context, in *ktpb.GetMonitoringRequest) (*ktpb.GetMonitoringResponse, error) {
+	// TODO(ismail): implement by revision API
+	return nil, grpc.Errorf(codes.Unimplemented, "GetSignedMapRoot is unimplemented")
+}
+
+func (s *Server) pollMutations(ctx context.Context, opts ...grpc.CallOption) ([]*ktpb.Mutation, error) {
+	req := &ktpb.GetMutationsRequest{PageSize: pageSize, Epoch: s.nextRevToQuery()}
+	resp, err := s.client.GetMutations(ctx, req, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	seen := time.Now().UnixNano()
+	if got, want := resp.GetSmr(), s.lastSeenSMR(); bytes.Equal(got.GetRootHash(), want.GetRootHash()) &&
+		got.GetMapRevision() == want.GetMapRevision() {
+		// We already processed this SMR. Do not update seen SMRs. Do not scroll
+		// pages for further mutations. Return empty mutations list.
+		glog.Infof("Already processed this SMR with revision %v. Continuing.", got.GetMapRevision())
+		return nil, nil
+	}
+
+	mutations := make([]*ktpb.Mutation, pageSize*2)
+	mutations = append(mutations, resp.GetMutations()...)
+	if err := s.pageMutations(ctx, resp, mutations, opts...); err != nil {
```

Comment:
Is there a reason `pageMutations` isn't creating and returning the list of mutations inside of it? 
In Go, we try to avoid returning values through arguments.